merino/providers/amo/backends/dynamic.py

```diff
@@ -37,41 +39,61 @@ def __init__(self, api_url: str):
         self.api_url = api_url
         self.dynamic_data = {}
 
+    async def _fetch_addon(
+        self, client: AsyncClient, addon_key: SupportedAddon
+    ) -> dict[str, Any] | None:
+        """Fetch addon metadata via AMO. Return `None` on errors."""
+        try:
+            res = await client.get(f"{self.api_url}/{addon_key}", follow_redirects=True)
+            res.raise_for_status()
+
+            json_res = res.json()
+            icon = json_res["icon_url"]
+            rating = f"{(json_res['ratings']['average']):.1f}"
+            number_of_ratings = json_res["ratings"]["count"]
+        except httpx.HTTPError:
+            logger.error(f"Addons API could not find key: {addon_key}")
+            return None
+        except (KeyError, JSONDecodeError):
+            logger.error(
+                "Problem with Addons API formatting. "
+                "Check that the API response structure hasn't changed."
+            )
+            return None
+
+        return {
+            "icon": icon,
+            "rating": rating,
+            "number_of_ratings": number_of_ratings,
+        }
```

Comment:
Suggestion for better flow-of-control

```suggestion
            json_res = res.json()
            return {
                "icon": json_res["icon_url"],
                "rating": f"{(json_res['ratings']['average']):.1f}",
                "number_of_ratings": json_res["ratings"]["count"],
            }
        except httpx.HTTPError:
            logger.error(f"Addons API could not find key: {addon_key}")
        except (KeyError, JSONDecodeError):
            logger.error(
                "Problem with Addons API formatting. "
                "Check that the API response structure hasn't changed."
            )

        return None
```